examples/eth_bots/execute_agent_trades.py

```diff
@@ -98,53 +109,86 @@ def transact_and_parse_logs(
     return trade_result
 
 
-def execute_agent_trades(
+async def async_execute_single_agent_trade(
+    account: EthAccount,
     web3: Web3,
     hyperdrive_contract: Contract,
-    agent_accounts: list[EthAccount],
+    hyperdrive_market: HyperdriveMarket,
 ) -> None:
-    """Hyperdrive forever into the sunset.
+    """Executes a single agent's trade. This function is async as
+    `match_contract_call_to_trade` waits for a transaction receipt
 
     Arguments
     ---------
     web3 : Web3
         web3 provider object
     hyperdrive_contract : Contract
         Any deployed web3 contract
-    agent_accounts : list[EthAccount]
-        A list of EthAccount that are conducting the trades
+    account: EthAccount
+        The EthAccount that is conducting the trade
+    hyperdrive_market: HyperdriveMarket:
+        The hyperdrive market state
     """
-    # get latest market
-    hyperdrive_market = hyperdrive_interface.get_hyperdrive_market(web3, hyperdrive_contract)
-    for account in agent_accounts:
-        # This condition is unlikely;
-        # it would happen if they created an eth address but didn't associate it with an elfpy Agent
-        if account.agent is None:
-            continue
-        trades: list[elftypes.Trade] = account.agent.get_trades(market=hyperdrive_market)
-        for trade_object in trades:
-            logging.info(
-                "AGENT %s to perform %s for %g",
-                str(account.checksum_address),
-                trade_object.trade.action_type,
-                float(trade_object.trade.trade_amount),
-            )
-            wallet_deltas = match_contract_call_to_trade(
+    # This condition is unlikely;
+    # it would happen if they created an eth address but didn't associate it with an elfpy Agent
+    if account.agent is None:
+        return
```

Comment:
I feel like we might as well raise an assertion error here, right? The code assumes this is true throughout.

Reply:
Currently, the code here will return gracefully. This function isn't returning anything, and instead, updating state variables internally. Still want to raise an assertion error here?

Reply:
Leaving as is for now